context/app/static/js/pages/Vis/Vis.jsx

```diff
@@ -33,55 +34,100 @@ import useSearchData from 'js/hooks/useSearchData';
 
  */
 
-const donorRaceQuery = {
+const donorRaceSexQuery = {
   size: 0,
-  query: {
-    bool: {
-      filter: {
-        term: {
-          entity_type: 'donor',
-        },
-      },
-    },
-  },
   aggs: {
-    'mapped_metadata.race': {
-      terms: {
-        field: 'mapped_metadata.race.keyword',
+    composite_data: {
+      composite: {
+        sources: [
+          {
+            'mapped_metadata.race': {
+              terms: {
+                field: 'mapped_metadata.race.keyword',
+              },
+            },
+          },
+          {
+            'mapped_metadata.blood_type': {
+              terms: {
+                field: 'mapped_metadata.blood_type.keyword',
+              },
+            },
+          },
+        ],
+        size: 10000,
       },
     },
   },
 };
 
 function Vis() {
   const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
-  const { searchData } = useSearchData(donorRaceQuery, elasticsearchEndpoint, nexusToken);
+
+  const { searchData } = useSearchData(donorRaceSexQuery, elasticsearchEndpoint, nexusToken);
   if (!('aggregations' in searchData)) {
     return null;
   }
-  const { buckets } = searchData.aggregations['mapped_metadata.race'];
+  function getCount(buckets, bloodType, race) {
+    const filtered = buckets.filter(
+      (b) => b.key['mapped_metadata.blood_type'] === bloodType && b.key['mapped_metadata.race'] === race,
+    );
+    return filtered.length ? filtered[0].doc_count : 0;
+  }
+  function getKeyValues(buckets, key) {
+    return [...new Set(buckets.map((b) => b.key[key]))];
+  }
+  const { buckets } = searchData?.aggregations.composite_data;
+
+  const bloodTypes = getKeyValues(buckets, 'mapped_metadata.blood_type');
 
   return (
-    <Paper>
-      <StyledTableContainer>
-        <Table stickyHeader>
-          <TableHead>
-            <TableRow>
-              {buckets.map((bucket) => (
-                <HeaderCell key={bucket.key}>{bucket.key}</HeaderCell>
-              ))}
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            <TableRow>
-              {buckets.map((bucket) => (
-                <TableCell>{bucket.doc_count}</TableCell>
-              ))}
-            </TableRow>
-          </TableBody>
-        </Table>
-      </StyledTableContainer>
-    </Paper>
+    Object.keys(searchData).length && (
+      <Paper>
+        <DonorChart
+          donorQuery={donorRaceSexQuery}
+          xKey="mapped_metadata.blood_type"
+          yKey="mapped_metadata.race"
+          colorKeys={['White', 'Black or African American', 'Hispanic']}
+          colors={['rgb(255, 99, 132)', 'rgb(54, 162, 235)', 'rgb(75, 192, 192)']}
+          title="Blood Type and Race"
+        />
+        <StyledTableContainer>
+          <Table stickyHeader>
+            <TableHead>
```

Comment:
I'd suggest factoring this out into a reusable component, like you did for the chart... Perhaps it's one component that could render both?